src/WebJobs.Script/Description/FunctionGenerator.cs

```diff
@@ -56,10 +56,39 @@ public static Type Generate(string functionAssemblyName, string typeName, Collec
                     continue;
                 }
 
+                var retValue = function.Parameters.Where(x => x.Name == ScriptConstants.SystemReturnParameterName).FirstOrDefault();
+                var parameters = function.Parameters.Where(x => x != retValue).ToArray();
+
                 MethodBuilder methodBuilder = tb.DefineMethod(function.Name, MethodAttributes.Public | MethodAttributes.Static);
-                Type[] types = function.Parameters.Select(p => p.Type).ToArray();
+                Type[] types = parameters.Select(p => p.Type).ToArray();
                 methodBuilder.SetParameters(types);
-                methodBuilder.SetReturnType(typeof(Task));
+
+                Type innerReturnType = null;
+
+                if (retValue == null)
+                {
+                    methodBuilder.SetReturnType(typeof(Task));
+                }
+                else
+                {
+                    // It's critical to set the proper return type for the binder.
+                    innerReturnType = retValue.Type.GetElementType(); // builder added a MakeByRefType
```

Comment:
I don't understand the "builder added a" comment. Perhaps just remove the comment?